### ISSUE TITLE
This table layout is broken

### ISSUE THREAD

**Issue by jrmuizel** [servo/servo#12696]
http://people.mozilla.org/~acomminos/A_colorful_table_of_browser_engines.html

**Comment by jrmuizel** [comment]
https://mdn.mozillademos.org/en-US/docs/Web/CSS/border-collapse$samples/A_colorful_table_of_browser_engines?revision=1160464 is a working url for the same thing

**Comment by nox** [comment]
<img width="1136" alt="capture d ecran 2017-10-04 a 19 21 15" src="https://user-images.githubusercontent.com/123095/31189671-353a304e-a939-11e7-9e91-5e6286d57341.png">

**Comment by dralley** [comment]
I actually can't open that URL (the working one that @jrmuizel) posted at all

```
[dalley@localhost servo]$ python3 ./mach run https://mdn.mozillademos.org/en-US/docs/Web/CSS/border-collapse$samples/A_colorful_table_of_browser_engines?revision=1160464
[2020-04-01T20:52:16Z ERROR script::dom::bindings::error] Error at https://mdn.mozillademos.org/static/build/js/main.ab2a3aefd0ce.js:5:28175 navMainSearchForm is null

```

**Comment by CYBAI** [comment]
hmm, interesting, I also tried to visited the working url and I actually saw a 404 page 🤔 (but I can visit the page on Firefox)

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/6782666/78319451-27bbeb00-75a2-11ea-8176-66b2fba83092.png">

**Comment by CYBAI** [comment]
I found we're getting response header `"x-cache"` with `"Error from cloudfront"` but it's `"Hit from cloudfront"` in Firefox 👀 

```
[2020-04-03T02:57:19Z INFO  net::http_loader] GET request for https://mdn.mozillademos.org/en-US/docs/Web/CSS/border-collapse/A_colorful_table_of_browser_engines?revision=1160464
[2020-04-03T02:57:19Z INFO  net::http_loader]  - ("accept", "text/html, application/xhtml+xml, application/xml; q=0.9, */*; q=0.8")
[2020-04-03T02:57:19Z INFO  net::http_loader]  - ("accept-language", "en-US, en; q=0.5")
[2020-04-03T02:57:19Z INFO  net::http_loader]  - ("user-agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:72.0) Servo/1.0 Firefox/72.0")
[2020-04-03T02:57:19Z INFO  net::http_loader]  - ("accept-encoding", "gzip, deflate, br")
[2020-04-03T02:57:20Z DEBUG net::http_loader] Not notifying devtools (no request_id)
[2020-04-03T02:57:20Z INFO  net::http_loader] HTTP/2.0 response for https://mdn.mozillademos.org/en-US/docs/Web/CSS/border-collapse/A_colorful_table_of_browser_engines?revision=1160464
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("content-type", "text/html; charset=utf-8")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("cache-control", "max-age=0, no-cache, no-store, must-revalidate")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("content-language", "en-US")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("date", "Fri, 03 Apr 2020 02:54:10 GMT")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("expires", "Fri, 03 Apr 2020 02:54:10 GMT")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("server", "meinheld/1.0.1")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("strict-transport-security", "max-age=63072000")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-content-type-options", "nosniff")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-frame-options", "DENY")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-xss-protection", "1; mode=block")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("content-encoding", "gzip")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("vary", "Accept-Encoding,Cookie")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-cache", "Error from cloudfront")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("via", "1.1 6ea9c3201cba3107c8908dce0b8278e4.cloudfront.net (CloudFront)")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-amz-cf-pop", "NRT20-C4")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("x-amz-cf-id", "HiwlitW91UEEzFO5-7jNmWkFF4o-F2NGMCC7GG8WO0kUTFu9PSCCdg==")
[2020-04-03T02:57:20Z INFO  net::http_loader]  - ("age", "189")
[2020-04-03T02:57:20Z DEBUG net::http_loader] got 404 response for "https://mdn.mozillademos.org/en-US/docs/... (843041cb3b6af48d)"
[2020-04-03T02:57:20Z INFO  net::hsts] adding host mdn.mozillademos.org to the strict transport security list
[2020-04-03T02:57:20Z INFO  net::hsts] - max-age 63072000
[2020-04-03T02:57:20Z DEBUG net::http_loader] successfully finished response for "https://mdn.mozillademos.org/en-US/docs/... (843041cb3b6af48d)"
```

**Comment by jdm** [comment]
@CYBAI Could you file that as a seperate issue, please?

**Comment by CYBAI** [comment]
Okay! 🙏

**Comment by CYBAI** [comment]
So, this page is actually still reproducible.

<img width="1027" alt="image" src="https://user-images.githubusercontent.com/6782666/78348613-4db2b100-75dd-11ea-9143-ec57d9ea118b.png">

(See https://github.com/servo/servo/issues/26099#issuecomment-608348117 for more details for the 404 page)

tl;dr: remember to run the page url with `'` when the url contains dollar sign

```
$ ./mach run 'https://mdn.mozillademos.org/en-US/docs/Web/CSS/border-collapse$samples/A_colorful_table_of_browser_engines?revision=1160464'
```

**Comment by mrobinson** [comment]
The test case from the Wayback machine:

```html
<style type="text/css">
.collapse {
  border-collapse: collapse;
}

.separate {
  border-collapse: separate;
}

table {
  display: inline-table;
  margin: 1em;
  border: dashed 5px;
}

table th,
table td {
  border: solid 3px;
}

.fx { border-color: orange blue; }
.gk { border-color: black red; }
.ed { border-color: blue gold; }
.tr { border-color: aqua; }
.sa { border-color: silver blue; }
.wk { border-color: gold blue; }
.ch { border-color: red yellow green blue; }
.bk { border-color: navy blue teal aqua; }
.op { border-color: red; }
</style>

<table class="separate">
  <caption><code>border-collapse: separate</code></caption>
  <tbody>
    <tr><th>Browser</th> <th>Layout Engine</th></tr>
    <tr><td class="fx">Firefox</td> <td class="gk">Gecko</td></tr>
    <tr><td class="ed">Edge</td> <td class="tr">EdgeHTML</td></tr>
    <tr><td class="sa">Safari</td> <td class="wk">Webkit</td></tr>
    <tr><td class="ch">Chrome</td> <td class="bk">Blink</td></tr>
    <tr><td class="op">Opera</td> <td class="bk">Blink</td></tr>
  </tbody>
</table>

<table class="collapse">
  <caption><code>border-collapse: collapse</code></caption>
  <tbody>
    <tr><th>Browser</th> <th>Layout Engine</th></tr>
    <tr><td class="fx">Firefox</td> <td class="gk">Gecko</td></tr>
    <tr><td class="ed">Edge</td> <td class="tr">EdgeHTML</td></tr>
    <tr><td class="sa">Safari</td> <td class="wk">Webkit</td></tr>
    <tr><td class="ch">Chrome</td> <td class="bk">Blink</td></tr>
    <tr><td class="op">Opera</td> <td class="bk">Blink</td></tr>
  </tbody>
</table>

**Comment by lumiscosity** [comment]
Appears to have been fixed. Tested on Arch Linux x86_64, servo built against commit id 9d29017c0dc6eec4ca7cd868cc13056e0bc0c64a, compared against Firefox 141.0, Epiphany 48.5 and Chromium 138.0.7204.168 (ordered left to right).

<img width="1920" height="466" alt="Image" src="https://github.com/user-attachments/assets/8c41894e-70bf-4fe3-88de-d55d78fdaaee" />

**Comment by jdm** [comment]
Excellent!
